Taskfile.yml

```
@@ -0,0 +1,25 @@
+version: 3
+
+silent: true
+
+tasks:
+  build_and_publish:
+    desc: Build and publish the module to Gem Fury
+    cmds:
+      - |
+        PUSH_URL=https://${GEM_FURY_PUSH_TOKEN}@push.fury.io/offerzenservices/
+
+        echo "Building gem..."
+        OUTPUT=$(gem build "${GEM_NAME}" | tail -1)
+        PACKAGE=$(echo $OUTPUT | cut -d\  -f2)
```

I think using the quoted space may be an easier "read" than using the escape char? Also, all `echo`s should be quoted to avoid [globbing and splitting](https://github.com/koalaman/shellcheck/wiki/SC2086). 😄
```suggestion
        PACKAGE=$(echo "$OUTPUT" | cut -d' ' -f2)
```

---

As an FYI as well for the other vars, you don't need to surround with `{ }` unless you're doing `bash` functions :) it's the equivalent of using an f-string in Python, but not using any refs in the string itself; you'll get the same output, but it's not "best-practice". (And yes, I think I've done this in the past...so I'm not innocent 😅)

---

Is there a way to prevent globbing and splitting and do the cut? `PACKAGE=$(echo "$OUTPUT" | cut -d' ' -f2)` doesn't work because I guess there's something that the splitting does that allows cut to work? I will find a better way to add the quotes and do the cut, thanks for raising.

---

I read a suggestion to use awk instead of cut because cut doesn't play nicely with quoted variables and delimiting by spaces because of the IFS.

awks 🤣